cypress/pageobjects/image.po.ts

```diff
@@ -1,45 +1,183 @@
+  url?: string,
+  size?: string,
+  path?: string,
+  labels?: any,
 }
 
-export default class ImagePage extends PagePo {
-  static url: string = `${Cypress.env('baseUrl')}dashboard/c/local/harvester/harvesterhci.io.virtualmachineimage`
+export class ImagePage {
+  name() {
+    return new LabeledInputPo('.namespace-select > .labeled-input', `:contains("Name")`)
+  }
+
+  url() {
+    return new LabeledInputPo('.labeled-input', `:contains("URL")`);
+  }
 
-  constructor() {
-    super(ImagePage.url);
+  description() {
+    return new LabeledInputPo('.labeled-input', `:contains("Description")`)
   }
 
   public goToList() {
-    cy.visit(ImagePage.url);
+    // cy.get('.nav').contains('Images').click();
+    cy.visit(constants.imagePage);
     cy.intercept('GET', '/v1/harvester/harvesterhci.io.virtualmachineimages').as('goToImageList');
-    cy.wait('@goToImageList')
-    expect(cy.url().should('eq', ImagePage.url));
+    cy.wait('@goToImageList');
+    cy.url().should('eq', constants.imagePage);
   }
 
   public goToCreate() {
-    this.goToList()
-    cy.contains('Create').click()
+    this.goToList();
+    cy.contains('Create').click();
+  }
+
+  public goToEdit(name: string) {
+    this.goToList();
+    cy.get('.search').type(name);
+    const image = cy.contains(name);
+
+    image.should('be.visible')
+    image.parentsUntil('tbody', 'tr').find('.icon-actions').click();
+    cy.get('.list-unstyled.menu').contains('Edit Config').click();
+  }
+
+  goToDetail(name: string) {
+    const image = cy.contains(name);
+    image.should('be.visible');
+    image.click();
+  }
+
+  public addLabels(value: ValueInterface) {
+    if (!value.labels) {
+      return;
+    }
+
+    const keys = Object.keys(value.labels);
+
+    cy.get('.tab#labels').click();
+    keys.forEach((key, index) => {
+      cy.contains('Add Label').click();
+      cy.get('.kv-item.key input').eq(index).type(key);
+      cy.get('.kv-item.value textarea').eq(index).type(value.labels[key]);
+    });
+  }
+
+  public create(value: ValueInterface) {
+    this.setValue(value);
+    if (value?.labels) {
+      this.addLabels(value);
+    }
+    this.save();
+  }
+
+  public edit(value: ValueInterface, upload: boolean = false) {
+    if (!upload) {
+      this.url().self().find('input').should('be.disabled')
+    }
+
+    this.description().input(value?.description);
+    this.addLabels(value);
+    this.save({ edit: true });
+  }
+
+  public createFromUpload(value: ValueInterface) {
+    this.setValue(value);
+    cy.get('.radio-group').contains('File').click();
+    cy.get('#file').selectFile(value.path, { force: true });
+    this.addLabels(value);
+    this.save({ upload: true });
   }
 
-  public save() {
-    cy.intercept('POST', '/v1/harvester/kubevirt.io.virtualmachineimage/*').as('createImage');
-    cy.get('.cru-resource-footer').contains('Create').click()
+  public exportImage(vmName: string, imageName: string) {
+    volumes.goToList();
+    cy.get('.search').type(vmName);
```

Comment:
The selector should be stored as a class member not selected just based on a CSS class
```
private search = '.search';
cy.get(this.search).type(vmName);
```

Reply:
done